packages/astro/src/content/types-generator.ts

```diff
@@ -118,13 +120,12 @@ export async function createContentTypesGenerator({
 
 			return { shouldGenerateTypes: true };
 		}
-		const entryInfo = getEntryInfo({
-			entry: event.entry,
-			contentDir: contentPaths.contentDir,
-		});
-		// Not a valid `src/content/` entry. Silently return.
-		if (entryInfo instanceof Error) return { shouldGenerateTypes: false };
 		if (fileType === 'unknown') {
+			const entryInfo = getEntryInfo({
+				entry: event.entry,
+				contentDir: contentPaths.contentDir,
+				allowFilesOutsideCollection: true,
```

Comment:
Is this intended? I'm not sure I quite follow the logic here being unfamiliar with this code part

Reply:
Ha, thought that would make this code _more_ readable instead of less 😅 
In short: we'll allow underscore `_` files outside of a collection (aka files with the `unknown `type). If that's unclear looking at the context, I can refactor another way.

Reply:
Oh no, it's fine. I just was missing some context on the changes being done here.

I do believe some comments would maybe make this a bit more readable, but that's outside the scope of this PR, in my opinion

Reply:
Gotcha, comment added 👍